projects/opendr_ws/src/opendr_perception/scripts/hr_pose_estimation_node.py

```diff
@@ -0,0 +1,164 @@
+from opendr.perception.pose_estimation import HighResolutionPoseEstimationLearner
+
+
+class PoseEstimationNode:
+
+    def __init__(self, input_rgb_image_topic="/usb_cam/image_raw",
+                 output_rgb_image_topic="/opendr/image_pose_annotated", detections_topic="/opendr/poses", device="cuda",
+                 num_refinement_stages=2, use_stride=False, half_precision=False):
+        """
+        Creates a ROS Node for high resolution pose estimation with HR Pose Estimation.
+        :param input_rgb_image_topic: Topic from which we are reading the input image
+        :type input_rgb_image_topic: str
+        :param output_rgb_image_topic: Topic to which we are publishing the annotated image (if None, no annotated
+        image is published)
+        :type output_rgb_image_topic: str
+        :param detections_topic: Topic to which we are publishing the annotations (if None, no pose detection message
+        is published)
+        :type detections_topic:  str
+        :param device: device on which we are running inference ('cpu' or 'cuda')
+        :type device: str
+        :param num_refinement_stages: Specifies the number of pose estimation refinement stages are added on the
+        model's head, including the initial stage. Can be 0, 1 or 2, with more stages meaning slower and more accurate
+        inference
+        :type num_refinement_stages: int
+        :param use_stride: Whether to add a stride value in the model, which reduces accuracy but increases
+        inference speed
+        :type use_stride: bool
+        :param half_precision: Enables inference using half (fp16) precision instead of single (fp32) precision.
+        Valid only for GPU-based inference
+        :type half_precision: bool
+        """
+        self.input_rgb_image_topic = input_rgb_image_topic
+
+        if output_rgb_image_topic is not None:
+            self.image_publisher = rospy.Publisher(output_rgb_image_topic, ROS_Image, queue_size=1)
+        else:
+            self.image_publisher = None
+
+        if detections_topic is not None:
+            self.pose_publisher = rospy.Publisher(detections_topic, OpenDRPose2D, queue_size=1)
+        else:
+            self.pose_publisher = None
+
+        self.bridge = ROSBridge()
+
+        # Initialize the high resolution pose estimation learner
+        self.pose_estimator = HighResolutionPoseEstimationLearner(device=device, num_refinement_stages=num_refinement_stages,
+                                                                  mobilenet_use_stride=use_stride,
+                                                                  half_precision=half_precision)
+        self.pose_estimator.download(path=".", verbose=True)
+        self.pose_estimator.load("openpose_default")
+
+    def listen(self):
+        """
+        Start the node and begin processing input data.
+        """
+        rospy.init_node('opendr_pose_estimation_node', anonymous=True)
```

Comment:
Should we rename this to hr_pose_estimtation_node?

Reply:
Yes it should be `opendr_hr_pose_estimation_node` preferably.

Reply:
I don't believe this was addressed yet

Reply:
I just checked the node and I see that it has been renamed 

https://github.com/opendr-eu/opendr/blob/hr_pose_estimation/projects/opendr_ws/src/opendr_perception/scripts/hr_pose_estimation_node.py#L82